content_sync/pipelines/definitions/concourse/site_pipeline.py

```diff
@@ -174,6 +174,23 @@ def __init__(  # noqa: PLR0913
         self.ocw_hugo_themes_branch = ocw_hugo_themes_branch
         self.ocw_hugo_projects_url = site.starter.ocw_hugo_projects_url
         self.ocw_hugo_projects_branch = ocw_hugo_projects_branch
+        if self.is_root_website:
+            self.base_url = ""
+            self.url_path = ""
+            self.delete_flag = ""
+            self.static_resources_subdirectory = f"/{site.get_url_path()}/"
+        else:
+            self.base_url = site.get_url_path()
+            self.static_resources_subdirectory = "/"
+            self.delete_flag = " --delete"
+        if self.is_test_website:
```

Comment:
Here if we are building a pipeline for a test site, we override all the necessary variables here to ensure that:

 - The site is not indexed by search engines
 - The site is always published to the test S3 buckets
 - The OCW Studio URL, resource base URL, sitemap domain and static api URL are all self-referential to the test site